spring-cloud-gcp-starters/spring-cloud-gcp-starter-sql/src/main/java/org/springframework/cloud/gcp/sql/autoconfig/GcpCloudSqlAutoConfiguration.java

```diff
@@ -127,8 +138,59 @@ public DataSource cloudSqlDataSource(CloudSqlJdbcInfoProvider cloudSqlJdbcInfoPr
 		hikariConfig.setUsername(this.gcpCloudSqlProperties.getUserName());
 		hikariConfig.setPassword(this.gcpCloudSqlProperties.getPassword());
 		// Setting this is useful to disable connection initialization. Especially in unit tests.
-		hikariConfig.setInitializationFailFast(this.gcpCloudSqlProperties.getInitFailFast());
+		hikariConfig.setInitializationFailFast(this.gcpCloudSqlProperties.isInitFailFast());
 
 		return new HikariDataSource(hikariConfig);
 	}
+
+	/**
+	 * Set credentials to be used by the Google Cloud SQL socket factory.
+	 *
+	 * <p>
+	 * The only way to pass a {@link CredentialFactory} to the socket factory is by passing a class
```

Comment:
It's a bit strong to say that it's the *only* way because you can also use a static variable or an environment variable, or some other global store.